### ISSUE TITLE
Run data_process.py

### ISSUE THREAD

**Issue by lylyjy** [xincoder/GRIP#14]
When I run data_process.py,
it says:

Generating Training Data.
(0,) (0,) (0,)
Generating Testing Data.
(0,) (0,) (0,)

I don't know the reason.

**Comment by xincoder** [comment]
Hi @lylyjy , thank you for your interest in our work. Please check your data_process.py, you should correct <data_root> [line 8] so that "prediction_train/*.txt" and "prediction_test/*.txt" are located in your <data_root> [line 185-186].

**Comment by lylyjy** [comment]
> Hi @lylyjy , thank you for your interest in our work. Please check your data_process.py, you should correct <data_root> [line 8] so that "prediction_train/_.txt" and "prediction_test/_.txt" are located in your <data_root> [line 185-186].

It doesn't work. The problem is it always outputs (0,).
In your computer, is it work?

**Comment by xincoder** [comment]
> > Hi @lylyjy , thank you for your interest in our work. Please check your data_process.py, you should correct <data_root> [line 8] so that "prediction_train/_.txt" and "prediction_test/_.txt" are located in your <data_root> [line 185-186].
> 
> It doesn't work. The problem is it always outputs (0,).
> In your computer, is it work?

Yes. It works on my side. The code should work, many researchers have already successfully run it. I tested the code on macOS and Ubuntu with python3.6 (it should also work for other python versions). Please check your python environment and downloaded data. (Download data from the ApolloScape website -> unzip the data -> modify the <data_root> accordingly and run the code. No extra operations are needed.) You should see the following output:
      > Generating Training Data.
      > (5010, 11, 12, 120) (5010, 120, 120) (5010, 2)
      > Generating Testing Data.
      > (415, 11, 6, 120) (415, 120, 120) (415, 2)

**Comment by lylyjy** [comment]
I have a question:  the problem occurs in the following code.
train_file_path_list = sorted(glob.glob(os.path.join(data_root, 'prediction_train/*.txt')))
test_file_path_list = sorted(glob.glob(os.path.join(data_root, 'prediction_test/*.txt')))

train_file_path_list is empty.
I need to replace * with the true document name?
Thanks.

**Comment by xincoder** [comment]
@lylyjy no, you do not need to do such a modification. "*" means any character (or string) here. Please refer to the glob official document for more information. Thanks.

**Comment by Hdjdnsj** [comment]
I have a question here: the "trajectory" dataset in Apollo website has three different types such as Sample Data, Training Data and Testing Data . Which should  I download? I am a rookie in this filed, I dont know how to modify the 'data root' if I download the Dataset in my location . It will be my pleasure if you can help me. Thanks. And my email adress is 1846389418@qq.com . You can send me too if in your convenience.

**Comment by Hdjdnsj** [comment]
你会了吗请问》我也有同样的问题，可以跟您联系一下吗？我的qq是1846389418，十分感谢

**Comment by Hdjdnsj** [comment]
> When I run data_process.py, it says:
> 
> Generating Training Data. (0,) (0,) (0,) Generating Testing Data. (0,) (0,) (0,)
> 
> I don't know the reason.

你会了吗请问？我也有同样的问题，可以跟您联系一下吗？我的qq是1846389418，十分感谢

**Comment by jay-tun** [comment]
> ```
> I have a question:  the problem occurs in the following code.
> ```
> 
> train_file_path_list = sorted(glob.glob(os.path.join(data_root, 'prediction_train/_.txt'))) test_file_path_list = sorted(glob.glob(os.path.join(data_root, 'prediction_test/_.txt')))
> 
> train_file_path_list is empty. I need to replace * with the true document name? Thanks.

Hi @lylyjy, @Hdjdnsj  , you need to have dataset prediction_test and prediction_train folder inside the data_root specified path. You can download those from https://github.com/ApolloScapeAuto/dataset-api/tree/master/trajectory_prediction or by following method: 

wget https://ad-apolloscape.cdn.bcebos.com/trajectory/prediction_train.zip
wget https://ad-apolloscape.cdn.bcebos.com/trajectory/prediction_test.zip

I recently managed to solve this issue as well. Hope that helps.
